server/logical.sql

```diff
@@ -0,0 +1,69 @@
+-- IF EXISTS(SELECT * FROM   dbo.users) DROP TABLE dbo.users;
+-- IF EXISTS(SELECT * FROM   dbo.cohort) DROP TABLE dbo.cohort;
+-- IF EXISTS(SELECT * FROM   dbo.events) DROP TABLE dbo.events;
+-- IF EXISTS(SELECT * FROM   dbo.recipes) DROP TABLE dbo.recipes;
+-- IF EXISTS(SELECT * FROM   dbo.dietary_restrictions) DROP TABLE dbo.dietary_restrictions;
+-- IF EXISTS(SELECT * FROM   dbo.allergies) DROP TABLE dbo.allergies;
+-- IF EXISTS(SELECT * FROM   dbo.recipe_allergies) DROP TABLE dbo.recipe_allergies;
+
+
+CREATE TABLE users(
+    id SERIAL PRIMARY KEY,
+    user_name VARCHAR(60) NOT NULL,
+    user_email VARCHAR(90) NOT NULL,
+    is_admin BOOLEAN NOT NULL,
+    is_lunch_maker BOOLEAN NOT NULL,
+    is_lunch_shopper BOOLEAN NOT NULL,
+    user_location VARCHAR(120) NOT NULL,
+    cohort_id INT REFERENCES cohort(id),
+    user_password VARCHAR(30) NOT NULL
+);
+
+CREATE TABLE cohort(
+    id SERIAL PRIMARY KEY,
+    class_number INT NOT NULL,
+    region VARCHAR(30) NOT NULL
+);
+
+CREATE TABLE events(
+    id SERIAL PRIMARY KEY,
+    meeting_location VARCHAR(60),
```

Comment:
Or you can have more column for the event table of meeting location, like Postcode (15), address_1(50), county(50), city(50),,,and so on.

Reply:
- **Event table**
you may end up with data duplication, as you may have different kind of events in the same meeting location, If agree u may take a look my suggestion else...
**Suggestion** try to create separate event_location table, with reference to the event table